src/v/utils/retry_chain_node.cc

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/retry_chain_node.h"
+
+#include "ssx/sformat.h"
+#include "vassert.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/ragel.hh>
+
+#include <limits>
+#include <variant>
+
+using namespace std::chrono_literals;
+
+/// Jitter
+thread_local static fast_prng _retry_rand_jitter_source;
+thread_local static uint32_t _logger_fiber_cnt = 0;
```

Comment:
can you add a comment on this `_logger_fiber_cnt` notice that we *only* use `_foo` leader under score for member variables.